src/stdatamodels/jwst/datamodels/tests/test_keyword_schemas.py

```diff
@@ -0,0 +1,137 @@
+        "core": ("core.schema", "meta.instrument.detector"),
+    },
+    "exptype": {
+        "keyword": ("keyword_exptype.schema", "meta.exposure.type"),
+        "pattern": ("keyword_pexptype.schema", "meta.exposure.p_exptype"),
+        "core": ("core.schema", "meta.exposure.type"),
+    },
+    "filter": {
+        "keyword": ("keyword_filter.schema", "meta.instrument.filter"),
+        "pattern": ("keyword_pfilter.schema", "meta.instrument.p_filter"),
+        "core": ("core.schema", "meta.instrument.filter"),
+    },
+    "grating": {
+        "keyword": ("keyword_grating.schema", "meta.instrument.grating"),
+        "pattern": ("keyword_pgrating.schema", "meta.instrument.p_grating"),
+        "core": ("core.schema", "meta.instrument.grating"),
+    },
+    "lampmode": {
+        "keyword": ("keyword_lampmode.schema", "meta.instrument.lamp_mode"),
+        # no p*
+        "core": ("core.schema", "meta.instrument.lamp_mode"),
+    },
+    # - lampstate -
+    # keyword_lampstate.schema.yaml single enum meta.instrument.lamp_state
+    # np p*
+    # not in core
+    "module": {
+        "keyword": ("keyword_module.schema", "meta.instrument.module"),
+        # no p*
+        "core": ("core.schema", "meta.instrument.module"),
+    },
+    "name": {
+        "keyword": ("referencefile.schema", "meta.instrument.name"),
+        # no p*
+        "core": ("core.schema", "meta.instrument.name"),
+    },
+    "noutputs": {
+        "keyword": ("keyword_noutputs.schema", "meta.exposure.noutputs"),
+        # no p*
+        "core": ("core.schema", "meta.exposure.noutputs"),
+    },
+    "pupil": {
+        "keyword": ("keyword_pupil.schema", "meta.instrument.pupil"),
+        "pattern": ("keyword_ppupil.schema", "meta.instrument.p_pupil"),
+        "core": ("core.schema", "meta.instrument.pupil"),
+    },
+    "readpatt": {
+        "keyword": ("keyword_readpatt.schema", "meta.exposure.readpatt"),
+        "pattern": ("keyword_preadpatt.schema", "meta.exposure.p_readpatt"),
+        "core": ("core.schema", "meta.exposure.readpatt"),
+    },
+    "subarray": {
+        "keyword": ("subarray.schema", "meta.subarray.name"),
+        "pattern": ("keyword_psubarray.schema", "meta.subarray.p_subarray"),
+        "core": ("core.schema", "meta.subarray.name"),
+    },
+}
+
+SCHEMA_URI_BASE = "http://stsci.edu/schemas/jwst_datamodel"
+
+
+def parse_pattern(regex_string):
+    return {i.strip("\\ ") for i in regex_string.strip("^(").split(")")[0].split("|")}
+
+
+def parse_schema_items(subschema):
+    # enum -> one enum
+    # pattern -> regex
+    # anyOf -> multiple enum
+    if "enum" in subschema:
+        return set(subschema["enum"])
+    if "pattern" in subschema:
+        return parse_pattern(subschema["pattern"])
+    if "anyOf" in subschema:
+        return set.union(*[parse_schema_items(i) for i in subschema["anyOf"]])
+    raise Exception("Not supported")  # noqa: TRY002
```

Comment:
I'm fine with the noqa here because it doesn't seem sensible to make a custom error type in a testing module, but could you make the error message more descriptive?

Reply:
How does the new message in 88591ea look?
```python
raise Exception(f"Failed to parse subschema {subschema}")  # noqa: TRY002
```